scripts/degenApe.js

```diff
@@ -115,301 +74,123 @@ const executeTx = async (calls, tx, fn, ...args) => {
+        console.log(`✔️ Successfully approved Jar to spend want`);
+      } else {
+        console.log(`Jar for this want already exists`);
+      }
 
       // Approve Strategy
-      await executeTx(callAttempts, 'approveStratTx', Controller.approveStrategy, txRefs['want'], txRefs['strategy'].address);
+      txRefs.'approveStratTx' = await executeTx(callAttempts, 'approveStratTx', Controller.approveStrategy, txRefs.'want', txRefs.'strategy'.address);
       console.log(`Strategy Approved!`);
 
-      // Set Jar
-      await executeTx(callAttempts, 'setJarTx', Controller.setJar, txRefs['want'], txRefs['jar'].address);
-      console.log(`Jar Set!`);
-
       // Set Strategy
-      await executeTx(callAttempts, 'setStratTx', Controller.setStrategy, txRefs['want'], txRefs['strategy'].address);
+      txRefs.'setStratTx' = await executeTx(callAttempts, 'setStratTx', Controller.setStrategy, txRefs.'want', txRefs.'strategy'.address);
       console.log(`Strategy Set!`);
       console.log(`✔️ Controller params all set!`);
 
       // Deposit Want
       console.log(`Depositing in Jar...`);
-      await executeTx(callAttempts, 'depositTx', txRefs['jar'].depositAll)
+      txRefs.'depositTx' = await executeTx(callAttempts, 'depositTx', txRefs.'jar'.depositAll)
       console.log(`✔️ Successfully deposited want in Jar`);
 
       // Call Earn
       console.log(`Calling earn...`);
-      await executeTx(callAttempts, 'earnTx', txRefs['jar'].earn);
+      txRefs.'earnTx' = await executeTx(callAttempts, 'earnTx', txRefs.'jar'.earn);
       console.log(`✔️ Successfully called earn`);
 
       //Push Strategy to be verified
-      testedStrategies.push(txRefs['strategy'].address)
+      testedStrategies.push(txRefs.'strategy'.address)
 
       // Call Harvest
-      console.log(`Waiting for ${sleepTime * 4 / 1000} seconds before harvesting...`);
-      await sleep(sleepTime * 4);
-      await executeTx(callAttempts, 'harvestTx', txRefs['strategy'].harvest);
-
-      await sleep(sleepTime, sleepToggle);
-      txRefs['ratio'] = await txRefs['jar'].getRatio();
-
-      if (txRefs['ratio'].gt(BigNumber.from(parseEther("1")))) {
-        console.log(`✔️ Harvest was successful, ending ratio of ${txRefs['ratio'].toString()} `);
+      console.log(`Waiting for ${ sleepConfigs.sleepTime * 4 / 1000 } seconds before harvesting...`);
+      await sleep(sleepConfigs.sleepTime * 4);
+      txRefs.'harvestTx' = await executeTx(callAttempts, 'harvestTx', txRefs.'strategy'.harvest);
+
+      await sleep(sleepConfigs);
+      txRefs.'ratio' = await txRefs.'jar'.getRatio();
+
+      if (txRefs.'ratio'.gt(BigNumber.from(parseEther("1")))) {
+        console.log(`✔️ Harvest was successful, ending ratio of ${ txRefs.'ratio'.toString() } `);
+
+        //Pf-core Generation
+        if (generatePfcore) {
+          const regex = /(?<=\$).*?(?=-)/g;
```

Comment:
Regex should usually be commented in english so people can have an idea of what pattern you're trying to match